app/controllers/post_favorites_controller.rb

```diff
@@ -0,0 +1,28 @@
+require_relative '../logical/danbooru/paginator/elasticsearch_extensions'
+class PostFavoritesController < ApplicationController
+  before_action :member_only
+  respond_to :html
+
+  def index
+    @post = Post.find(params[:post_id])
+    # order by most recent favorites first
+    all_favs = @post.favorited_users_by_id.reverse
+    if params.has_key? :page
+      page = params[:page].to_i || 1
+    else
+      page = 1
```

Comment:
Somehow this is the only place in code where this is done which makes me think theres probably a way to remove this. The problem I had was that if no page query parameter was present it was converted to 0 instead of using the code after ||

Reply:
`[params[:page].to_i, 1].max` is a more Ruby-like way to do it (and also handles more annoying cases)

Reply:
Thank you, that looks much cleaner. That also guarantees your page will always be 1 or more. Previously you would get a 404 not found for that, but I guess if you're putting negative pages your asking for it.

I force-pushed the change